### DISCUSSION TITLE
[Feature]   Is it possible to add an button for external library to button auto Remove offline File

### DISCUSSION THREAD

**Discussion by li1220** [immich-app/immich#9300] [Feature Request]
### I have searched the existing feature requests to make sure this is not a duplicate request.

- [X] Yes

### The feature

 Is it possible to add an button for external library to button auto Remove offline File

----------------------------------------------------
能否给外部库添加一个按钮来控制是否自动删除脱机文件？

### Platform

- [X] Server
- [ ] Web
- [ ] Mobile

**Comment by bo0tzz** [comment]
This already exists (https://immich.app/docs/features/libraries#deleted-external-assets). If that's not what you mean, can you elaborate?

**Reply by li1220** [reply]
Currently, you can only manually click the "Remove Offline Files" button to delete offline files. My idea is to add a button to control whether to automatically delete offline files without confirming whether the external library is offline or the file has been deleted.

**Reply by li1220** [reply]
Because my immich and the external library are on the same server (NAS), there will be no situation where the external library goes offline alone.

**Comment by li1220** [comment]
Can such a feature be added?

**Reply by bo0tzz** [reply]
We deliberately don't support it because it can cause problems with misconfigured mounts and such. You can use a cronjob to call the api https://immich.app/docs/api/remove-offline-files
